src/fswAlgorithms/attDetermination/inertialAttitudeUkf/inertialAttitudeUkf.cpp

```diff
@@ -0,0 +1,303 @@
+                /*! - Apply low-pass filter to gyro measurements to get smoothed body rate*/
+                lowPass.processMeasurement(omega_BN_B);
+                }
+
+            auto gyroMeasurement = Measurement();
+            gyroMeasurement.name = "gyro";
+            gyroMeasurement.timeTag = meanMeasurementTime;
+            gyroMeasurement.validity = true;
+            gyroMeasurement.size = 3;
+
+            gyroMeasurement.noise = this->gyroNoise/std::sqrt(numberOfValidGyroMeasurements);
+            gyroMeasurement.observation = lowPass.getCurrentState();
+            gyroMeasurement.model = lastThreeStates;
+            this->measurements[this->measurementIndex] = gyroMeasurement;
+            this->measurementIndex += 1;
+            }
+}
+
+/*! Read the message containing the measurement data.
+ * It updates class variables relating to measurement data including validity and time tags.
+ @return void
+ */
+void InertialAttitudeUkf::readFilterMeasurements() {
+    /*! - Read static RW and spacecraft config data message and store it in module fields */
+    if (this->firstFilterPass) {
+        this->rwArrayConfigPayload = this->rwArrayConfigMsg();
+        auto vehicleConfigPayload = this->vehicleConfigMsg();
+
+        this->spacecraftInertia = cArray2EigenMatrix3d(vehicleConfigPayload.ISCPntB_B);
+        this->spacecraftInertiaInverse = this->spacecraftInertia.inverse();
+    }
+
+    this->measurementIndex = 0;
+    /*! - Read in wheel speeds, their time, and compute the wheel accelerations for the propagation method*/
+    readRWSpeedData();
+    /*! - Read star tracker measurements*/
+    readStarTrackerData();
+    /*! Only add the gyro measurements to processing if the filter is in a mode that desires that */
+    if (this->measurementAcceptanceMethod == AttitudeFilterMethod::AllMeasurements) {
+        readGyroData();
+    }
+    if (measurementAcceptanceMethod == AttitudeFilterMethod::GyroWhenDazzled && !this->validStarTracker) {
+        readGyroData();
+    }
+}
+
+/*! Integrate the equations of motion of two body point mass gravity using Runge-Kutta 4 (RK4)
+    @param interval integration interval
+    @param X0 initial state
+    @param dt time step
+    @return Eigen::VectorXd
+*/
+Eigen::VectorXd InertialAttitudeUkf::propagate(std::array<double, 2> interval, const Eigen::VectorXd& X0, double dt){
+    double t_0 = interval[0];
+    double t_f = interval[1];
+    double t = t_0;
+    Eigen::VectorXd X = X0;
+
+    std::function<Eigen::VectorXd(double, Eigen::VectorXd)> attitudeDynamics = [this](double t, Eigen::VectorXd state)
+    {
+        Eigen::Vector3d mrp(state.head(3));
+        Eigen::Vector3d omega(state.tail(3));
+        Eigen::MatrixXd Bmat = bmatMrp(mrp);
+
+        Eigen::VectorXd stateDerivative(state.size());
+        stateDerivative.head(3) << 0.25*Bmat*omega;
+
+        Eigen::Vector3d wheelTorque;
```

Comment:
:large_blue_circle: This could be declared and initialized in one expression using `Eigen::Vector3d wheelTorque = Eigen::Vector3d::Zero();`

Reply:
Done